swap/peer.go

```diff
@@ -30,18 +35,144 @@ var ErrDontOwe = errors.New("no negative balance")
 // Peer is a devp2p peer for the Swap protocol
 type Peer struct {
 	*protocols.Peer
+	lock               sync.RWMutex
 	swap               *Swap
 	beneficiary        common.Address
 	contractAddress    common.Address
 	lastReceivedCheque *Cheque
+	lastSentCheque     *Cheque
+	balance            int64
 }
 
 // NewPeer creates a new swap Peer instance
 func NewPeer(p *protocols.Peer, s *Swap, beneficiary common.Address, contractAddress common.Address) *Peer {
-	return &Peer{
+	peer := &Peer{
 		Peer:            p,
 		swap:            s,
 		beneficiary:     beneficiary,
 		contractAddress: contractAddress,
 	}
+
+	peer.lastReceivedCheque, _ = s.loadLastReceivedCheque(p.ID())
+	peer.lastSentCheque, _ = s.loadLastSentCheque(p.ID())
+	peer.balance, _ = s.loadBalance(p.ID())
+
+	return peer
+}
+
+func (p *Peer) getLastReceivedCheque() *Cheque {
```

Comment:
I wonder if we need functions for these accesses if we just return the object right away....

Reply:
I based this on viktor's comment on the incentives PR (https://github.com/ethersphere/swarm/pull/1554#discussion_r314620178) where they were functions.